invenio_records_files/links.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2016 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Link for file bucket creation."""
+
+from flask import url_for
+from invenio_deposit.api import Deposit
+
+
+def default_bucket_link_factory(pid):
+    """Factory for record bucket generation."""
+    if not hasattr(pid, 'get_assigned_object'):
+        return None
+
+    deposit = Deposit.get_record(pid.get_assigned_object())
```

Comment:
I used `Deposit` instead of `Record` because it provides an implementation for `_create_bucket`, which is called if a bucket doesn't already exist when the `files` property in `FilesMixin` is called. Should I use `Record` anyway and replicate the `_create_bucket` behavior in the case a bucket doesn't exist?